src/library/autodj/autodjfeature.cpp

```diff
@@ -308,8 +310,22 @@ void AutoDJFeature::slotAddRandomTrack(bool) {
         retriveAttempts += 1;
     }
 
-    if (retriveAttempts == kMaxRetiveAttempts)
-        qDebug () << "Could Not Load Random Tracks";
+    if (retriveAttempts == kMaxRetiveAttempts || iTrackId == -1) {
+        qDebug () << "Could Not Load Random Tracks From Crate."
+    	        " Attempting to load from library";
+    	iTrackId = m_autoDjCratesDao.getRandomTrackIdFromLibrary();
```

Comment:
This is starting to get complex as well. Please consider to tweak the autoDJ track source table for the no crate use case. Not sure, which way will work best. 
https://github.com/mixxxdj/mixxx/blob/38dcd763c2ffc46e6958ca687a8a83e928059c4d/src/library/dao/autodjcratesdao.cpp#L105